magellan-legacy/src/main/java/com/wealthfront/magellan/Screen.java

```diff
@@ -37,21 +46,23 @@
  * }
  * </code> </pre>
  */
-public abstract class Screen<V extends ViewGroup & ScreenView> implements BackHandler {
+public abstract class Screen<V extends ViewGroup & ScreenView> extends LifecycleAwareComponent implements Navigable {
 
   public static final int DEFAULT_ACTION_BAR_COLOR_RES = 0;
   private static final String VIEW_STATE = "com.wealthfront.navigation.Screen.viewState";
 
+  private final DialogComponent dialogComponent = new DialogComponent();
+
   private Activity activity;
-  private Navigator navigator;
```

Comment:
This is going to break a lot of code without a migration. If we want to do this, we should provide an easy-to-use script that will update all `Screen`s with whatever new pattern we recommend, otherwise this will be painful.

Reply:
Added back the navigator.